### PR TITLE
fix: graceful exit

### PR DESCRIPTION
**Pull request by MCozhusheck**
Description
---
fixes: #662 
Exiting app during initialization (especially during connecting to other peers) caused app to run some processes in the background.

Motivation and Context
---
Fix issue describe above.

How Has This Been Tested?
---
Run TU and exit during initialization while looking for other peers in the network.

What process can a PR reviewer use to test or verify this change?
---
Same as above

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

**Comment by MCozhusheck** [comment]
> Tested locally but this does not solve the problem

Found and fixed and another commit issue with p2pool locking RWLock which caused dangling p2pool process. It was due p2pool trying to get status from status monitor which was already killed.

File: src-tauri/src/process_watcher.rs [hunk 1]
```diff
@@ -68,7 +68,9 @@ impl<TAdapter: ProcessAdapter> ProcessWatcher<TAdapter> {
             loop {
                 select! {
                       _ = watch_timer.tick() => {
+                            info!(target: LOG_TARGET, "Checking if {} is running", name);
```
Comment (stringhandler) [inline src-tauri/src/process_watcher.rs:71]:
```suggestion
                            debug!(target: LOG_TARGET, "Checking if {} is running", name);
```

File: src-tauri/src/process_watcher.rs [hunk 2]
```diff
@@ -68,7 +68,9 @@ impl<TAdapter: ProcessAdapter> ProcessWatcher<TAdapter> {
...
                             if child.ping() {
+                                info!(target: LOG_TARGET, "{} is running", name);
```
Comment (stringhandler) [inline src-tauri/src/process_watcher.rs:73]:
```suggestion
                                debug!(target: LOG_TARGET, "{} is running", name);
```

File: src-tauri/src/p2pool_manager.rs [hunk 3]
```diff
@@ -147,11 +147,12 @@ impl P2poolManager {
             .await?;
         process_watcher.wait_ready().await?;
         if let Some(status_monitor) = &process_watcher.status_monitor {
-            loop {
+            for attempt in 1..=5 {
                 sleep(Duration::from_secs(5)).await;
```
Comment (stringhandler) [inline src-tauri/src/p2pool_manager.rs:151]: 5 seconds is very long here, rather make it 1 second but run more attempts

**Comment by brianp** [comment]
Not working for me. The base node is still being left orphaned after I've shut down the app.

image

**Comment by stringhandler** [comment]
Think this is fixed in 0.5.20

**Pull request closed without merging**